services/s3/src/main/resources/codegen-resources/customization.config

```diff
@@ -6,6 +6,17 @@
     "Error": "S3Error",
     "Object": "S3Object"
   },
+  "shapeAdditions": {
+    "shapes":{
+      "PartType":{
+        "type":"string",
+        "enum":[
```

Comment:
Hmm, this does not seem to be flexible to work with other types. Should we use fqcn to load the custom class?

Reply:
Added synthetic field and made all custom shapes synthetic. Modified ShapeModelSpec so that synthetic shapes are excluded from SdkFields